app/src/androidTest/java/org/mozilla/focus/activity/screenshots/NotificationScreenshots.java

```diff
@@ -0,0 +1,108 @@
+import tools.fastlane.screengrab.Screengrab;
+import tools.fastlane.screengrab.locale.LocaleTestRule;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.pressImeActionButton;
+import static android.support.test.espresso.action.ViewActions.replaceText;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.hasFocus;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static junit.framework.Assert.assertTrue;
+import static org.hamcrest.Matchers.containsString;
+
+/**
+ * This test has been super flaky in the past so we moved it to its own test. This way we wat least
+ * only lose screenshots of this test in case it fails.
+ */
+@RunWith(AndroidJUnit4.class)
+public class NotificationScreenshots extends ScreenshotTest {
+    @Rule
+    public ActivityTestRule<MainActivity> mActivityTestRule = new MainActivityFirstrunTestRule(false);
+
+    @ClassRule
+    public static final LocaleTestRule localeTestRule = new LocaleTestRule();
+
+    private MockWebServer webServer;
+
+    @Before
+    public void setUpWebServer() throws IOException {
+        webServer = new MockWebServer();
+
+        // Test page
+        webServer.enqueue(new MockResponse().setBody(TestHelper.readTestAsset("plain_test.html")));
+    }
+
+    @After
+    public void tearDownWebServer() {
+        try {
+            webServer.close();
+            webServer.shutdown();
+        } catch (IOException e) {
+            throw new AssertionError("Could not stop web server", e);
+        }
+    }
+
+    @Test
+    public void takeScreenshotOfNotification() throws Exception {
+        onView(withId(R.id.url_edit))
+                .check(matches(isDisplayed()))
+                .check(matches(hasFocus()))
+                .perform(click(), replaceText(webServer.url("/").toString()), pressImeActionButton());
+
+        onView(withId(R.id.display_url))
+                .check(matches(isDisplayed()))
+                .check(matches(withText(containsString(webServer.getHostName()))));
+
+        final UiObject openAction = device.findObject(new UiSelector()
+                .descriptionContains(getString(R.string.notification_action_open))
+                .resourceId("android:id/action0")
+                .enabled(true));
+
+        device.openNotification();
+
+        try {
+            if (!openAction.waitForExists(waitingTime)) {
+                // The notification is not expanded. Let's expand it now.
+                device.findObject(new UiSelector()
+                        .text(getString(R.string.app_name)))
+                        .swipeDown(5);
```

Comment:
I noticed low step number of .swipeDown() can cause unpredictable results, try experimenting with double digit numbers (for me 50 seems pretty safe)

Reply:
Oh, I forgot to update this PR. Locally I was using 50. On some emulators it seems to trigger a long-press which is super awkward.